doc/rtd/reference/datasources/none.rst

```diff
@@ -0,0 +1,51 @@
+.. _datasource_none:
+
+None
+****
+
+The data source ``None`` may be used when no other viable datasource is
+present on disk. This has three primary use cases:
+
+1. Testing cloud-init in an environment where no datasource is
+   available.
+2. As a fallback for when a datasource is otherwise intermittently
+   unavailable.
+3. Providing user data to cloud-init from on-disk configuration when
+   no other datasource is present.
+
+When the datasource is ``None``, cloud-init is unable to obtain or
+render networking configuration. Additionally, when cloud-init
+completes, a warning is logged that DataSourceNone is being used.
+
+Configuration
+=============
+
+User data and meta data may be passed to cloud-init via system
+configuration in :file:`/etc/cloud/cloud.cfg` or
+:file:`/etc/cloud/cloud.cfg.d/`.
```

Comment:
The files in `cloud.cfg.d/` require a specific file ending that is not a standard *.yaml/*.yml, right? I know ds-identify code does require that, at least.

Do you think that is worth implying with `/etc/cloud/cloud.cfg.d/*.cfg`?